src/components/date-range-picker/date-range-picker.ts

```diff
@@ -0,0 +1,1211 @@
+    return this._activeDate ?? this._calendar?.activeDate;
+  }
+
+  /**
+   * Whether to show the number of the week in the calendar.
+   * @attr show-week-numbers
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'show-week-numbers' })
+  public showWeekNumbers = false;
+
+  /**
+   * Controls the visibility of the dates that do not belong to the current month.
+   * @attr hide-outside-days
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'hide-outside-days' })
+  public hideOutsideDays = false;
+
+  /** Gets/sets special dates. */
+  @property({ attribute: false })
+  public specialDates!: DateRangeDescriptor[];
+
+  /** Sets the start day of the week for the calendar. */
+  @property({ attribute: 'week-start' })
+  public weekStart: WeekDays = 'sunday';
+
+  // #endregion
+
+  // #endregion
+
+  // #region Life-cycle hooks
+
+  constructor() {
+    super();
+    this._formValue = createFormValueState<DateRangeValue | null>(this, {
+      initialValue: {
+        start: null,
+        end: null,
+      },
+      transformers: defaultDateRangeTransformers,
+    });
+
+    this._rootClickController.update({ hideCallback: this._handleClosing });
+    this.addEventListener('focusin', this._handleFocusIn);
+    this.addEventListener('focusout', this._handleFocusOut);
+
+    addKeybindings(this, {
+      skip: () => this.disabled || this.readOnly,
+      bindingDefaults: { preventDefault: true },
+    })
+      .set([altKey, arrowDown], this.handleAnchorClick)
+      .set([altKey, arrowUp], this._onEscapeKey)
+      .set(escapeKey, this._onEscapeKey);
+  }
+
+  protected override firstUpdated() {
+    this._setCalendarRangeValues();
+    this._delegateInputsValidity();
+  }
+
+  protected override formResetCallback() {
+    super.formResetCallback();
+    this._setCalendarRangeValues();
+    this._updateMaskedRangeValue();
+  }
+
+  // #endregion
+
+  // #region Public API methods
+
+  /** Clears the input parts of the component of any user input */
+  public clear() {
+    this._oldValue = this.value;
+    this.value = null;
+    if (this.useTwoInputs) {
+      this._inputs[0]?.clear();
+      this._inputs[1]?.clear();
+    }
+  }
+
+  /** Selects a date range value in the picker */
+  public select(value: DateRangeValue | null) {
+    this._select(value);
+  }
+
+  // #endregion
+
+  // #region Observed properties
+
+  @watch('open')
+  protected _openChange() {
+    this._rootClickController.update();
+
+    if (this.open) {
+      this._oldValue = this.value;
+    }
+  }
+
+  @watch('locale')
+  protected _updateDefaultMask(): void {
+    this._defaultMask = DateTimeUtil.getDefaultMask(this.locale);
+    this._updateMaskedRangeValue();
+  }
+
+  @watch('useTwoInputs')
+  protected async _updateDateRange() {
+    await this._calendar?.updateComplete;
+    this._updateMaskedRangeValue();
+    this._setCalendarRangeValues();
+  }
+
+  @watch('mode')
+  protected async _modeChanged() {
+    if (this.mode === 'dialog') {
```

Comment:
```suggestion
    if (!this._isDropDown) {
```
I see there are a couple of checks and you already have a utility to check it